src/AnswersHeadlessProvider.tsx

```diff
@@ -1,19 +1,38 @@
 import { ReactChild, ReactChildren } from 'react';
 import { provideAnswersHeadless, AnswersHeadless, AnswersConfig } from '@yext/answers-headless';
 import { AnswersHeadlessContext } from './AnswersHeadlessContext';
+import { v4 as uuidv4 } from 'uuid';
 
-interface Props extends AnswersConfig {
+type Props = AnswersConfig & {
   children?: ReactChildren | ReactChild | (ReactChildren | ReactChild)[],
-  verticalKey?: string
-}
+  verticalKey?: string,
+  sessionTrackingEnabled?: boolean
+};
 
 export function AnswersHeadlessProvider(props: Props): JSX.Element {
-  const { children, verticalKey, ...answersConfig } = props;
+  const { children, verticalKey, sessionTrackingEnabled=true, ...answersConfig } = props;
   const answers: AnswersHeadless = provideAnswersHeadless(answersConfig);
   verticalKey && answers.setVerticalKey(verticalKey);
+  handleSessionTracking(answers, sessionTrackingEnabled);
```

Comment:
Rather than having `handleSessionTracking`, we could abstract out the window session storage code into a method called `acquireSessionId`. I'm thinking `acquire` rather than `get` because there may be the side effect of setting the session Id, and I think `acquire` captures that better. So the code would look like:

```
answers.setSessionTrackingEnabled(sessionTrackingEnabled);
if (sessionTrackingEnabled) {
    const sessionId = acquireSessionId();
    answers.setSessionId(sessionId);
}

This allows us to avoid having a `handleSessionTracking` function operating at different levels of abstraction. Answers headless is very high-level, and the sessionStorage API is very low level.

Reply:
sounds good. Yeah `acquire` is better in this context. updated